src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotRaiseExceptionsInUnexpectedLocations.cs

```diff
@@ -329,6 +332,21 @@ private static bool IsEqualityOperator(IMethodSymbol method, Compilation compila
             };
         }
 
+        private static bool IsComparisonOperator(IMethodSymbol method, Compilation compilation)
+        {
+            if (!method.IsStatic || !method.IsPublic())
+                return false;
+
+            return method.Name switch
+            {
+                WellKnownMemberNames.LessThanOperatorName
+                or WellKnownMemberNames.GreaterThanOperatorName
+                or WellKnownMemberNames.LessThanOrEqualOperatorName
+                or WellKnownMemberNames.GreaterThanOrEqualOperatorName => true,
+                _ => false,
+            };
+        }
```

Comment:
nit: Per spec, `==` and `!=` are also comparison operators. So you might want to consider merging `IsComparisonOperatror` and `IsEqualityOperator` into one method `IsComparisonOperator`.

https://github.com/dotnet/csharplang/blob/main/spec/expressions.md#integer-comparison-operators

![image](https://user-images.githubusercontent.com/31348972/118472668-5dba2100-b709-11eb-9144-0dafad08dbbb.png)

Feel free if you don't want to do that since @mavasani has already approved.

Reply:
I went ahead and fixed it. It's a simple change.